src/compiler/checker.ts

```diff
@@ -167,10 +167,10 @@ namespace ts {
 
         return checker;
 
-        function getEmitResolver(sourceFile?: SourceFile) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {

```

Comment:
Why not pass the cancellation token to createTypeChecker from the program. the program has the cancelation token, then the checker can hold on to it.